kamon-core/src/main/resources/reference.conf

```diff
@@ -150,45 +150,202 @@ kamon {
-    # Binary transports. Only broadcast keys configured bellow will be processed by the context Codec. The FQCN of
-    # the appropriate Codecs for each key must be provided, otherwise keys will be ignored.
-    #
-    codecs {
-
-      # Size of the encoding buffer for the Binary Codec.
-      binary-buffer-size = 256
-
-      # Declarative definition of broadcast context keys with type Option[String]. The setting key represents the actual
-      # key name and the value is the HTTP header name to be used to encode/decode the context key. The key name will
-      # be used when coding for binary transport. The most common use case for string keys is effortless propagation of
-      # correlation keys or request related data (locale, user ID, etc). E.g. if wanting to propagate a "X-Request-ID"
-      # header this config should suffice:
-      #
-      # kamon.context.codecs.string-keys {
-      #   request-id = "X-Request-ID"
-      # }
+      # Default HTTP propagation. Unless specified otherwise, all instrumentation will use the configuration on
+      # this section for HTTP context propagation.
       #
-      # If the application must read this context key they can define key with a matching name and read the value from
-      # the context:
-      #   val requestIDKey = Key.broadcastString("request-id") // Do this only once, keep a reference.
-      #   val requestID = Kamon.currentContext().get(requestIDKey)
-      #
-      string-keys {
+      default {
+
+        # Configures how context tags will be propagated over HTTP headers.
+        #
+        tags {
+
+          # Header name used to encode context tags.
+          header-name = "context-tags"
+
+
+          # Provide explicit mappins between context tags and the HTTP headers that will carry them. When there is
+          # an explicit mapping for a tag, it will not be included in the default context header. For example, if
+          # you wanted to use the an HTTP header called `X-Correlation-ID` for a context tag with key `correlationID`
+          # you would need to include a the following configuration:
+          #
+          #   mappings {
+          #     correlationID = "X-Correlation-ID"
+          #   }
+          #
+          # The correlationID tag would always be read and written from the `X-Correlation-ID` header. The context
+          # tag name is represented as the configuration key and the desired header name is represented by the
+          # cofiguration value.
+          #
+          mappings {
+
+          }
+        }
 
+        # Configure which entries should be read from incoming HTTP requests and writen to outgoing HTTP requests.
+        #
+        entries {
+
+          # Specify mappings between Context keys and the Propagation.EntryReader[HeaderReader] implementation in charge
+          # of reading them from the incoming HTTP request into the Context.
+          incoming {
```

Comment:
I think that would be a good idea support (in a future) multiples  Readers/Writers for the same key  for example:
```
 incoming {
       span =[ "kamon.trace.SpanPropagation$B3", "kamon.trace.SpanPropagation$B3Single"]
 }
```
maybe after merge https://github.com/kamon-io/Kamon/pull/551 WDYT?